lib/new_relic/agent/method_tracer_helpers.rb

```diff
@@ -109,6 +116,22 @@ def namespace_and_location(object, method_name)
         end
         [name, method.source_location, is_class_method]
       end
+
+      # Rails controllers can be a special case because by default, controllers in Rails
+      # automatically render views with names that correspond to valid routes. This means
+      # that a controller method may not have a corresponding method in the controller class.
+      def controller_without_method?(klass, method_name)
+        defined?(ApplicationController) && klass < ApplicationController && !klass.method_defined?(method_name)
+      end
+
+      def controller_namespace_and_location(klass, method_name, is_class_method, name)
+        return [] unless defined?(Rails)
+
+        path = Rails.root.join("app/controllers/#{klass.name.underscore}.rb")
```

Comment:
This makes an assumption we're looking at a Rails app without nested controllers (haven't found a way to determine the file location of a class). If this fails, we at least will be muting the false warnings.

Reply:
Have you tried assuming namespacing leads to nesting? For example, if I saw `Users::LoginController` I would look for the file in `app/controllers/users/login_controller.rb`

Reply:
Yep, @hannahramadan whipped up a classic admin controller with content in an "admin" subdirectory. For a given `Admin::UsersController` class, she found that calling `.name.underscore` on the class would correctly yield the desired `admin/users_controller` value.

Reply:
Oh I see - @kaylareopelle is responding to @hannahramadan's comment here more than the code. @hannahramadan you got nesting working. The only thing you don't have working from my vantage point is support for controllers living outside of `app/controllers`. I'm not sure what hackery is involved to offer that or how common it is, but I think that's ok for now.

Reply:
You got it, @fallwith! Thanks for providing some context and correcting my misreading of the initial comment. I also think it's okay to skip automatic detection for controllers that live outside `app/controllers`.

Reply:
@fallwith @kaylareopelle thanks both! My comment isn't accurate, thanks for correcting it ◡̈